examples/react-hooks-example/src/App.js

```diff
@@ -0,0 +1,134 @@
+import React, {useEffect, useState} from 'react';
+import {
+    Box,
+    Grommet,
+    DataTable,
+    Text
+} from 'grommet';
+import { pipe } from 'rxjs';
+import Subspace, {$latest} from '@embarklabs/subspace';
+import Web3 from 'web3';
+import exchangeABI from './contract/exchange_abi.json'
+
+const web3 = new Web3('https://cloudflare-eth.com');
+const subspace = new Subspace(web3.currentProvider);
+var dai = new web3.eth.Contract(exchangeABI, '0x2a1530C4C41db0B0b2bB646CB5Eb1A67b7158667');
+const daiContract = subspace.contract(dai);
+subspace.init();
+
+const theme = {
+    global: {
+        font: {
+            family: 'Roboto',
+            size: '18px',
+            height: '20px'
+        }
+    }
+};
+
+
+function App() {
+
+    const [txnObserver, setObservable] = useState();
+    const [last5Observer, setLast5Observer] = useState();
+    const [latestBlock, setBlock] = useState();
+    const [last5, setLast5] = useState([]);
+
+    //Trade details object for calculating exchange rate
+    function TradeDetails(tokensSold, ethBought) {
+        this.tokensSold = web3.utils.fromWei(tokensSold);
+        this.ethBought = web3.utils.fromWei(ethBought);
+        this.exchangeRate = this.tokensSold / this.ethBought;
+    }
+
+    //Effect hook to define subspace observables
+    useEffect(() => {
```

Comment:
I believe the hooks logic can be simplified a bit. You can do something like: 
```js

function App() {
    const [last5Observer, setLast5Observer] = useState();
    const [last5, setLast5] = useState([]);

    //Trade details object for calculating exchange rate
    function TradeDetails(tokensSold, ethBought) {
        this.tokensSold = web3.utils.fromWei(tokensSold);
        this.ethBought = web3.utils.fromWei(ethBought);
        this.exchangeRate = this.tokensSold / this.ethBought;
    }

    useEffect(() => {
      (async () => {
        const latestBlock = await web3.eth.getBlockNumber();
        const last5$ = daiContract.events.EthPurchase.track({
                         fromBlock: latestBlock - 50
                       }).pipe(
                         map(trade => {
                          const txnDetails = new TradeDetails(trade.tokens_sold, trade.eth_bought);
                          return {"block": trade.blockNumber, "rate": txnDetails.exchangeRate};
                        }),
                        $latest(5)
                       );
        setLast5Observer(last5$);
      })();
    });

    //Effect hook to source last 5 trades from EthPurchase observableyarn
    useEffect(() => {
        if (last5Observer === undefined) {
            return;
        } 
        const subscription = last5Observer.subscribe(setLast5);
    
        return () => subscription.unsubscribe();
    }, [last5Observer]);

    return (
        <Grommet theme={theme}>
          ...
    );
}

```

Reply:
Agree, you definitely could.  I had split it out to highlight the different tricks that can be done with Subspace (e.g. have one observable that's just tracking a contract event and then a completely different effect/observable to show the event sourcing/observable transformation).  If you'd rather it show this approach, I'm good either way.

Reply:
oh, for educational purposes I agree with your approach

Reply:
@acolytec3 -- might be a good idea to add this thought to the article so that users understand as well.